control-plane/api-gateway/binding/result.go

```diff
@@ -382,59 +386,78 @@ func (l listenerValidationResult) conflictedCondition(generation int64) metav1.C
 }
 
 // acceptedCondition constructs the condition for the ResolvedRefs status type.
-func (l listenerValidationResult) resolvedRefsCondition(generation int64) metav1.Condition {
+func (l listenerValidationResult) resolvedRefsConditions(generation int64) []metav1.Condition {
 	now := timeFunc()
 
+	conditions := make([]metav1.Condition, 0)
+
 	if l.routeKindErr != nil {
-		return metav1.Condition{
+		return []metav1.Condition{{
 			Type:               "ResolvedRefs",
 			Status:             metav1.ConditionFalse,
 			Reason:             "InvalidRouteKinds",
 			ObservedGeneration: generation,
 			Message:            l.routeKindErr.Error(),
 			LastTransitionTime: now,
-		}
+		}}
 	}
 
-	switch l.refErr {
-	case errListenerInvalidCertificateRef_NotFound, errListenerInvalidCertificateRef_NotSupported, errListenerInvalidCertificateRef_InvalidData, errListenerInvalidCertificateRef_NonFIPSRSAKeyLen, errListenerInvalidCertificateRef_FIPSRSAKeyLen:
-		return metav1.Condition{
-			Type:               "ResolvedRefs",
-			Status:             metav1.ConditionFalse,
-			Reason:             "InvalidCertificateRef",
-			ObservedGeneration: generation,
-			Message:            l.refErr.Error(),
-			LastTransitionTime: now,
-		}
-	case errRefNotPermitted:
-		return metav1.Condition{
-			Type:               "ResolvedRefs",
-			Status:             metav1.ConditionFalse,
-			Reason:             "RefNotPermitted",
-			ObservedGeneration: generation,
-			Message:            l.refErr.Error(),
-			LastTransitionTime: now,
+	for _, refErr := range l.refErrs {
+		switch refErr {
+		case errListenerInvalidCertificateRef_NotFound,
+			errListenerInvalidCertificateRef_NotSupported,
+			errListenerInvalidCertificateRef_InvalidData,
+			errListenerInvalidCertificateRef_NonFIPSRSAKeyLen,
+			errListenerInvalidCertificateRef_FIPSRSAKeyLen:
+			conditions = append(conditions, metav1.Condition{
+				Type:               "ResolvedRefs",
+				Status:             metav1.ConditionFalse,
+				Reason:             "InvalidCertificateRef",
+				ObservedGeneration: generation,
+				Message:            refErr.Error(),
+				LastTransitionTime: now,
+			})
```

Comment:
Not for this PR, but something to consider looping back around on: maybe we should have a function that creates `metav1.Condition` objects with the current timestamp given how much of this file is defining those objects. Then we could reduce the amount of code by having a function call that builds the condition.

```go
func newCondition(type string, status metav1.Status, reason string, generation int64, message: string) metav1.Condition {
    return metav1.Condition{
		Type:               type,
		Status:             status,
		Reason:             reason,
		ObservedGeneration: generation,
		Message:            message,
		LastTransitionTime: timeFunc(),
	}
}
```